### ISSUE TITLE
Gets rollback automatically while creating infrastructure

### ISSUE THREAD

**Issue by viku85** [awslabs/aws-api-gateway-developer-portal#488]
When using [api-gateway-dev-portal](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:563878140293:applications~api-gateway-dev-portal) it executes for a while and then gets rollback automatically.

![rollback-screenshot](https://user-images.githubusercontent.com/1793164/120663247-7a937a00-c4a7-11eb-890d-8470e02081bd.png)

Can someone help out with the reason or any guidelines to resolve it?

**Comment by amazon-meaisiah** [comment]
Could you provide more details about the error it's encountering? You should be able to look in the stack and see what resource failed as well as what error caused it to roll back the deployment.

**Comment by praveenrengarajan** [comment]
@amazon-meaisiah facing the same issue
![image](https://user-images.githubusercontent.com/2884636/122709649-0710ab80-d2a2-11eb-8e8f-264250bd59c3.png)

**Comment by amazon-meaisiah** [comment]
@praveenrengarajan Could you show me the precise error that first occurred before the rest of the deployment was rolled back?

I see the following were not created:

- DevPortalSiteS3Bucket
- CustomersTable
- CognitoUserPoolClientSettingsBackingFnRole
- CatalogUpdaterLambdaExecutionRole
- CloudFrontOriginAccessIdentity
- AssetUploaderLambdaExecutionRole
- LambdaEdgeFunctionRole
- CognitoUserPoolDomainBackingFnRole
- CloudFrontEdgeReplicatorRple
- LambdaCommonLayerb90953e8ec
- PreLoginAccountsTable
- CognitoPreSignUpTriggerExecutionRole

I see the following were cancelled:

- DevPortalSiteS3Bucket
- CognitoUserPoolClientSettingsBackingFnRole
- CatalogUpdaterLambdaExecutionRole
- CloudFrontOriginAccessIdentity
- AssetUploaderLambdaExecutionRole
- LambdaEdgeFunctionRole
- CognitoUserPoolDomainBackingFnRole
- CloudFrontEdgeReplicatorRple
- LambdaCommonLayerb90953e8ec
- PreLoginAccountsTable
- CognitoPreSignUpTriggerExecutionRole

What was the failure reason for CustomersTable?

**Comment by mbreuker** [comment]
I had a similar problem. When I tried again, I get:

`Stack:arn:aws:cloudformation:eu-central-1:658636592913:stack/serverlessrepo-api-gateway-dev-portal/602a23a0-e3cc-11eb-b969-062dc65bb180 is in ROLLBACK_COMPLETE state and can not be updated.`

**Comment by mbreuker** [comment]
I also get an automated rollback with this error:

![Schermafbeelding 2021-07-14 om 16 25 10](https://user-images.githubusercontent.com/131899/125638999-dc20765e-3567-49f6-837f-b3fc18921cf7.png)

**Comment by mbreuker** [comment]
Found the solution to my error in #425

**Comment by mbreuker** [comment]
> @amazon-meaisiah facing the same issue
> ![image](https://user-images.githubusercontent.com/2884636/122709649-0710ab80-d2a2-11eb-8e8f-264250bd59c3.png)
> Note: am using my personal AWS account (administrator)

You need to make sure that the S3 buckets that you specify, don't already exist. The first time I tried deploying the application I used existing buckets which caused a rollback.

**Comment by Jyoti-Singh-XFT** [comment]
I am still facing the same issue even after adding the fix mentioned in #425

**Comment by amazon-meaisiah** [comment]
@Jyoti-Singh-XFT Have you checked that the above S3 buckets don't already exist? If so, could you give some more details on the precise error you're running into (specifically, what's causing the rollbacks)?

**Comment by praveenrengarajan** [comment]
![image](https://user-images.githubusercontent.com/2884636/132349702-673247ee-a781-4ed9-b521-ba46a98384e6.png)

Where to update the config in the source code that all resources for dev-portal should be in ap-southeast-2 ??

**Comment by amazon-meaisiah** [comment]
@praveenrengarajan What command are you running to try to deploy it? If you're using SAM, you should be able to just use `sam package --region ap-southeast-2` + `sam deploy --region ap-southeast-2` and call it a day. If you're using `node run release` per the local development instructions, #505 should unblock you once it's merged.

**Comment by praveenrengarajan** [comment]
@amazon-meaisiah will  cloudfront lambda@edge deploy in ap-southeast-2 ?

**Comment by amazon-meaisiah** [comment]
No, it must be in us-east-1 [due to CloudFront restrictions](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/edge-functions-restrictions.html#lambda-at-edge-function-restrictions). The custom resource in the template used for creating this essentially exists only to work around [this SAM issue](https://github.com/aws/serverless-application-model/issues/635), where they don't really have special support for Lambda@Edge beyond a very rudimentary level.

**Comment by talipini** [comment]
Hi - I am getting exactly same errors. Any help?

**Comment by asheroto** [comment]
I'm having the same error, followed this verbatim, clicked on [https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:563878140293:applications~api-gateway-dev-portal](Deploy) (didn't use SAM), except that I got an error that says CognitoDomainNameOrPrefix is required so I set it.

Every time I deploy it it just rolls back.

![image](https://user-images.githubusercontent.com/49938263/176786832-3b1e4c88-895a-4bfd-b276-52907be8189a.png)
